src/eval.rs

```diff
@@ -344,7 +344,7 @@ pub fn eval_entry<'tcx>(
     })();
 
     // Machine cleanup.
-    EnvVars::cleanup(&mut ecx).unwrap();
+    ecx.allow_data_races_mut(|ecx| EnvVars::cleanup(ecx).unwrap());
```

Comment:
Please leave some comments explaining why we do this (maybe referencing this PR, too?)